f/connectors/alerts/alerts_gcs.py

```diff
@@ -321,17 +336,33 @@ def prepare_alerts_metadata(alerts_metadata, territory_id):
 
     Returns
     -------
-    list of dict
+    prepared_alerts_metadata : list of dict
         A list of dictionaries representing the filtered and processed alerts
         metadata, including additional columns for geolocation, metadata UUID,
         and alert source.
+    alerts_statistics : dict
+        A dictionary containing alert statistics: total alerts, month/year,
+        and description of alerts.
     """
     # Convert CSV bytes to DataFrame
     df = pd.read_csv(StringIO(alerts_metadata))
 
     # Filter DataFrame based on territory_id
     filtered_df = df.loc[df["territory_id"] == territory_id]
 
+    # Group the DataFrame by month and year, and get the last row for each group
+    filtered_df = (
+        filtered_df.loc[df["territory_id"] == territory_id]
+        .groupby(["month", "year"])
+        .last()
+        .reset_index()
+    )
+
+    # Sort the DataFrame by year and month in descending order
+    filtered_df = filtered_df.sort_values(
+        by=["year", "month"], ascending=[False, False]
+    )
```

Comment:
nit: I'd probably move this call to be in the same line as your `.iloc[0]` below.  That line **depends on** this having been run, and you do not want to risk someone changing one of the lines without the other.

Also I think nothing else in this file depends on this sorting besides the `.iloc[0]`, so you don't even necessarily need to persist the result of the sort.

Reply:
Thanks, good catch!